redwood-lazylayout-uiview/src/commonMain/kotlin/app/cash/redwood/lazylayout/uiview/UIViewLazyList.kt

```diff
@@ -160,69 +171,109 @@ internal open class UIViewLazyList(
-        UICollectionViewScrollPositionTop,
+        UITableViewScrollPosition.UITableViewScrollPositionTop,
         animated = false,
       )
     }
   }
 
-  override var modifier: Modifier = Modifier
+  override fun itemsBefore(itemsBefore: Int) {
+    processor.itemsBefore(itemsBefore)
+  }
+
+  override fun itemsAfter(itemsAfter: Int) {
+    processor.itemsAfter(itemsAfter)
+  }
 
-  override val value: UIView get() = collectionView
   override fun onEndChanges() {
-    collectionView.reloadData()
+    processor.onEndChanges()
   }
 }
 
 private const val reuseIdentifier = "LazyListContainerCell"
 
-private class LazyListContainerCell(frame: CValue<CGRect>) : UICollectionViewCell(frame) {
-  lateinit var collectionViewFrame: CValue<CGRect>
-  lateinit var collectionViewFlowLayout: UICollectionViewFlowLayout
+internal class LazyListContainerCell(
+  style: UITableViewCellStyle,
+  reuseIdentifier: String?,
+) : UITableViewCell(style, reuseIdentifier) {
+  internal var cell: LazyListUpdateProcessor.Cell<LazyListContainerCell, UIView>? = null
+  internal var widgetView: UIView? = null
+
+  override fun initWithStyle(
+    style: UITableViewCellStyle,
+    reuseIdentifier: String?,
+  ): UITableViewCell = LazyListContainerCell(style, reuseIdentifier)
+
+  override fun initWithFrame(
+    frame: CValue<CGRect>,
+  ): UITableViewCell {
+    return LazyListContainerCell(UITableViewCellStyle.UITableViewCellStyleDefault, null)
+      .apply { setFrame(frame) }
+  }
+
+  override fun willMoveToSuperview(newSuperview: UIView?) {
+    super.willMoveToSuperview(newSuperview)
+
+    // Confirm the cell is bound when it's about to be displayed.
+    if (superview == null && newSuperview != null) {
+      require(cell!!.isBound) { "about to display a cell that isn't bound!" }
+    }
+
+    // Unbind the cell when its view is detached from the table.
+    if (superview != null && newSuperview == null) {
+      removeAllSubviews()
+      cell?.unbind()
+      cell = null
+    }
+  }
 
-  private var widgetView: UIView? = null
-  override fun initWithFrame(frame: CValue<CGRect>): UICollectionViewCell = LazyListContainerCell(frame)
   override fun prepareForReuse() {
     super.prepareForReuse()
-    // clear out subviews here
-    this.contentView.subviews.forEach {
-      (it as UIView).removeFromSuperview()
-    }
-    widgetView = null
+    removeAllSubviews()
+    cell?.unbind()
+    cell = null
   }
 
-  fun set(widgetView: UIView) {
+  fun setWidget(widgetView: UIView) {
     this.widgetView = widgetView
+
+    removeAllSubviews()
     contentView.addSubview(widgetView)
-    contentView.layoutIfNeeded()
+    contentView.translatesAutoresizingMaskIntoConstraints = false
+    setNeedsLayout()
   }
 
   override fun layoutSubviews() {
     super.layoutSubviews()
-    widgetView?.setFrame(this.contentView.bounds)
+
+    val widgetView = this.widgetView ?: return
+    widgetView.setFrame(bounds)
```

Comment:
nit: since `widgetView` is a subview of `contentView`, I would probably do
```
contentView.setFrame(bounds)
widgetView.setFrame(contentView.bounds)
```
instead, but it should have the same effect

Reply:
we'll also definitely want to call `widgetView.sizeToFit()` here

Reply:
After playing with this I’ve learned that contentView is disinterested in delegating to its children for stuff like sizing, so the cell needs to control both. I’m gonna keep it as-is for now ’cause it seems to be working!